src/graph/savegraphaction.pas

```diff
@@ -23,7 +23,18 @@ interface
     '.png',
     '.jpg'
   );
-
+  // invalid characters in filenames for each OS
+  InvalidChars: TSysCharSet = [
```

Comment:
I'm almost 100% certain there is a native FPC method that can do this for us. Writing our own logic for handling OS paths is very prone to errors.

Reply:
Sort of, but it's a mess. (https://forum.lazarus.freepascal.org/index.php?topic=63675.0) I'm going to drop this whole idea and just use sequential numbers to name multiple charts. That is, just add 1,2,... to the name provided by the user. This guarantees that file names will be distinct. Using the strata values won't always yield distinct names if the values include invalid characters. We may still need the helper function to insert the sequential numbers. I'll keep it only if we need to create the file names in more than one place. (Can't easily check all the code right now). We can let the OS take exception to invalid file names provided by the user.

Reply:
No need for this. In my test, the OS causes an exception, which is caught by Analysis and reported to the user.